pkg/controller/metrics/metrics.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metrics
+
+import (
+	"context"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	log "github.com/sirupsen/logrus"
+
+	hivev1 "github.com/openshift/hive/pkg/apis/hive/v1alpha1"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+var (
+	metricClusterDeploymentsTotal = prometheus.NewGauge(prometheus.GaugeOpts{
+		Name: "hive_cluster_deployments_total",
+		Help: "Total number of cluster deployments that exist in Hive.",
+	})
+	metricClusterDeploymentsInstalledTotal = prometheus.NewGauge(prometheus.GaugeOpts{
+		Name: "hive_cluster_deployments_installed_total",
+		Help: "Total number of cluster deployments that are successfully installed.",
+	})
+)
+
+func init() {
+	metrics.Registry.MustRegister(metricClusterDeploymentsTotal)
+	metrics.Registry.MustRegister(metricClusterDeploymentsInstalledTotal)
+}
```

Comment:
@ShivamChamoli just as a note for you, the above is how you can create and register a metric in the cluster deployment controller.